### PR TITLE
[feat] Add dataset and data_loader builder

### PR DESCRIPTION
**Pull request by JanLin0817**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Develop DIR
 dev/
-datasets/*
```
Comment (mousyball) [inline .gitignore:3]: 'datasets/*' was used to soft link the dataset, such as voc, coco and etc.
We have to make sure where the place is used to put the dataset.

Probably 'dev/data' is an option as the code listed below.

File: configs/pytorch_trainer/epoch_trainer.yaml [hunk 2]
```diff
@@ -1,4 +1,7 @@
-_BASE_: [./hook/trainer_hook.yaml, ../networks/classification/lenet.yaml]
+_BASE_: [./hook/trainer_hook.yaml,
+         ../networks/classification/lenet.yaml,
+         ../datasets/cifar10.yaml,
+]
```
Comment (mousyball) [inline configs/pytorch_trainer/epoch_trainer.yaml]: The inheritance order of config should start from general components and end with network settings.
[Example]
Trainer.yaml should inherit from optimizer, logger, hook and so on.
(Just make an example. Eventually, the trainer includes all components setting in a file for simplicity.)
Then, the final 'custom_network.yaml' includes the default trainer config.

If there is any modification of trainer setting, just override it in the final 'custom_network.yaml'.

File: pytorch_trainer/trainer/base_trainer.py [hunk 3]
```diff
@@ -165,5 +165,9 @@ def register_callback(self,
         # self._register_hook(config.CUSTOM_HOOK)
 
     def data_to_device(self, data):
-        inputs, labels = data
-        return inputs.to(self.device), labels.to(self.device)
+        # TODO: place to hook
+        if isinstance(data, dict):
+            return {k: v.to(self.device) for k, v in data.items()}
+        else:
```
Comment (mousyball) [inline pytorch_trainer/trainer/base_trainer.py]: If we use dict output in dataloader, just keep dict version only.

**Review by mousyball — COMMENTED** [review]
OK

**Comment by mousyball** [comment]
Please review 2 additional updates.

* [misc] Modify the default of dataset path to dev/data
* [feat] Parse dict data in train/val step.

**Comment by mousyball** [comment]
Please review 1 additional updates.

* [feat] Transform builder

File: .gitignore (hunk 1 above)
Comment (JanLin0817) [reply .gitignore:3]: OK

File: pytorch_trainer/trainer/base_trainer.py (hunk 3 above)
Comment (JanLin0817) [reply pytorch_trainer/trainer/base_trainer.py]: ok

File: configs/custom/classification/lenet/epoch_trainer.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+_BASE_:
+  [
+    "../../../datasets/cifar10.yaml",
+    "../../../networks/classification/lenet.yaml",
+    "../../../pytorch_trainer/epoch_trainer.yaml",
+  ]
```
Comment (mousyball) [inline configs/custom/classification/lenet/epoch_trainer.yaml:6]: Inheritance Hierarchy is correct.
But the naming of leaf file should be the network info.
For example, `fcos_r50_caffe_fpn_gn-head_mstrain_640-800_2x_coco.yaml`.
But it doesn't have to be fixed.
Just make a note of naming logic.

**Review by mousyball — COMMENTED** [review]
GJ

**Pull request merged**